pre-processing/primod/driver_coupling/ribameta.py

```diff
@@ -0,0 +1,148 @@
+import copy
+from pathlib import Path
+from typing import Any
+
+import geopandas as gpd
+import imod
+import numpy as np
+import ribasim
+from imod.msw import GridData, MetaSwapModel, Sprinkling
+
+from primod.driver_coupling.driver_coupling_base import DriverCoupling
+from primod.driver_coupling.util import (
+    _nullify_ribasim_exchange_input,
+    _validate_node_ids,
+)
+from primod.mapping.svat_basin_mapping import SvatBasinMapping
+from primod.mapping.svat_user_demand_mapping import SvatUserDemandMapping
+
+
+class RibaMetaDriverCoupling(DriverCoupling):
+    """A dataclass representing one coupling scenario for the RibaMod driver.
+
+    Attributes
+    ----------
+    basin_definition: gpd.GeoDataFrame
+        GeoDataFrame of basin polygons
+    user_demand_definition: gpd.GeoDataFrame
+        GeoDataFrame of user demand polygons
+    """
+
+    ribasim_basin_definition: gpd.GeoDataFrame
+    ribasim_user_demand_definition: gpd.GeoDataFrame | None
```

Comment:
```suggestion
    ribasim_user_demand_definition: gpd.GeoDataFrame | None = None
```

You have to set a default value if you want to make it optional. Otherwise, you have to specify `None` when constructing it. If that's what you want you can ignore me

Reply:
Good catch!